compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -1666,24 +1665,33 @@ public void visit(BLangSimpleVarRef bLangSimpleVarRef) {
 
                 if (symbol instanceof BVarSymbol) {
                     ((BVarSymbol) symbol).originalSymbol = null;
-                    if (withinLambdaFunc && symbol.closure) {
-                        // When there's a closure in a lambda inside a query lambda the symbol.closure is
-                        // true for all its usages. Therefore mark symbol.closure = false for the existing
-                        // symbol and create a new symbol with the same properties.
-                        symbol.closure = false;
-                        symbol = new BVarSymbol(0, symbol.name, env.scope.owner.pkgID, symbol.type, env.scope.owner,
-                                pos, VIRTUAL);
-                        symbol.closure = true;
-                        bLangSimpleVarRef.symbol = symbol;
-                        bLangSimpleVarRef.varSymbol = symbol;
+                    if (withinLambdaFunc) {
+                        if (symbol.closure) {
+                            // When there's a closure in a lambda inside a query lambda the symbol.closure is
+                            // true for all its usages. Therefore mark symbol.closure = false for the existing
+                            // symbol and create a new symbol with the same properties.
```

Comment:
What's an example for this? Why are we changing the existing one and redefining a new one instead of using the new one altogether?

Reply:
We created a new symbol for variable references within the lambda function and set the `closure` flag as true.  (This change is not introduced with this PR. )

example
```ballerina
function testQueryExpWithinQueryAction() returns error? {
    int[][] data = [[2, 3, 4]];
    check from int[] arr in data
        do {
            function () returns int[] func = function() returns int[] {
                int[] evenNumbers = from int i in arr
                    where i % 2 == 0
                    select i;
                return evenNumbers;
            };
            int[] expected = [2, 4];
    };
}
```